test/mpi/testlist.gpu

```diff
@@ -0,0 +1,3 @@
+coll
+pt2pt
+rma
```

Comment:
Does this mean these directories will be executed twice, once in `testlist` and another time in `testlist.gpu`?

Reply:
No, in this case runtests is only executing `testlist.gpu` files.

Reply:
Oh, I see. In a default configuration they may execute twice. That isn't good.

Reply:
Yeah, this has bothered me. There are following solutions:
* Let `runtests` always descend to sub directories. Place specific "disable" entry in `testlist` to skip directory. For example: `#f77` skips `f77`.
* Let `runtests` to cache directories, so it can skip on repeats.

Reply:
> * Let `runtests` always descend to sub directories. Place specific "disable" entry in `testlist` to skip directory. For example: `#f77` skips `f77`. This simplifies custom tests, since we don't need populate entries at each upper directory.

Do you mean add a disable directive in f77/testlist?

Reply:
> Do you mean add a disable directive in f77/testlist?

No. I mean by default do not put dir entries in testlist. Only put `#[dir]` if we want to explicitly disable the entire directory. In the example, we put `#f77` in `test/mpi/testlist` to skip `f77` tests. There is no need to modify `f77/testlist`.

To make it convenient for configure, we may allow dir entries in testlist but ignores it -- since it will descend into the dir anyway.

Reply:
> In the example, we put `#f77` in `test/mpi/testlist` to skip `f77` tests.

I guess it might be better to explicitly use `f77 -skip`.

Reply:
We still need to enumerate all the test directories somewhere, though. Or are you saying descend into all directories, even those without `testlist` files?

Reply:
> Or are you saying descend into all directories, even those without `testlist` files?

Yes.